shell.nix

```diff
@@ -0,0 +1,58 @@
+# SPDX-FileCopyrightText: 2022 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+{ pkgs ? import nix/nixpkgs.nix {} }:
+pkgs.mkShell {
+  name = "shell";
+  buildInputs =
+    # Ideally, we'd add all (Hackage) dependencies to the dependency list so we
+    # don't have to compile everything ourselves. I haven't figured out how to
+    # do this yet without introducing conflicts with `cabal.project`.
+    #
+    # pkgs.haskellPackages.bittide-extra.env.nativeBuildInputs ++
+    #
+    [
```

Comment:
If I want to use this locally with a fresh nix install, then `pkgs.git` is required for me as well. Otherwise I get
```
Error: cabal: The program 'git' is required but it could not be found.
```
when running `cabal build`.

Reply:
I'll add it because I think CI needs it too when running with `--pure`, but for developers _not_ using `--pure` is perfectly fine.